sheet_to_triples/tests/test_field.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+import unittest
+
+from .. import field
+
+
+def _dt_from_str(value):
+    return datetime.datetime.strptime(value, '%Y-%m-%d')
+
+
+class CellTestCase(unittest.TestCase):
+
+    def test_as_slug(self):
+        self.assertEqual(
+            field.Cell('  Some?letters+<&>characters  ').as_slug,
+            'some-letters-and-characters'
+        )
+        with self.assertRaises(ValueError):
+            field.Cell('').as_slug
+
+    def test_as_uc(self):
+        self.assertEqual(
+            field.Cell('   Some?letters+<and>characters   ').as_uc,
+            'SomeLettersAndCharacters'
+        )
+        with self.assertRaises(ValueError):
+            field.Cell('').as_uc
+
+    def test_as_json(self):
+        self.assertEqual(
+            field.Cell({'list': ['é', 1, None]}).as_json,
+            '{"list":["é",1,null]}'
+        )
+        with self.assertRaises(ValueError):
+            field.Cell('').as_json
+
+    def test_as_text(self):
+        self.assertEqual(
+            field.Cell(1).as_text,
+            '1'
+        )
+        with self.assertRaises(ValueError):
+            field.Cell('   ').as_text
+
+    def test_as_capital(self):
+        self.assertEqual(
+            field.Cell('   shared   ').as_capital,
+            'vm:capitalTypes/sharedFinancial'
+        )
+        errors_values = (
+            (None, ValueError),
+            ('notfound', TypeError),
+            ('   ', ValueError),
+        )
+        for value, error in errors_values:
+            with self.assertRaises(error):
+                field.Cell(value).as_capital
```

Comment:
Generally I'd have a method for the success(es) and a method for the fail(s), rather than combined.

Also, have the option of using subtests for loops like this, for slightly better failure output:
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

So, would end up something like:
```python
    def test_as_capital(self):
        self.assertEqual(
            field.Cell('   shared   ').as_capital,
            'vm:capitalTypes/sharedFinancial'
        )

    ERRORS_CASES = (
        (None, ValueError),
        ('notfound', TypeError),
    )
    
    def test_as_capital_error(self):
        cases = ERRORS_CASES + (('  ', ValueError),)
        for value, error in cases:
           with self.subtest(cell=value):
                with self.assertRaises(error):
                    field.Cell(value).as_capital
```

Reply:
Ah yeah, that's exactly what I was trying to dredge up from my memory when looking for a better way of doing test feedback.